test/src/main/java/org/axonframework/test/saga/SagaTestFixture.java

```diff
@@ -585,15 +600,44 @@ public void injectResources(Object saga) {
                              .filter(f -> registeredResources.contains(ReflectionUtils.getFieldValue(f, saga)))
                              .findFirst()
                              .ifPresent(field -> {
-                                 throw new AssertionError(format("Field %s.%s is injected with a resource, " +
-                                                                         "but it doesn't have the 'transient' modifier.\n"
-                                                                         +
-                                                                         "Mark field as 'transient' or disable this check using:\n"
-                                                                         +
-                                                                         "fixture.withTransienceCheckDisabled()",
-                                                                 field.getDeclaringClass(), field.getName()));
+                                 throw new AssertionError(format(
+                                         "Field %s.%s is injected with a resource,"
+                                                 + " but it doesn't have the 'transient' modifier."
+                                                 + "\nMark field as 'transient' or disable this check using:"
+                                                 + "\nfixture.withTransienceCheckDisabled()",
+                                         field.getDeclaringClass(),
+                                         field.getName()
+                                 ));
                              });
             }
         }
     }
+
+    /**
+     * Wrapping {@link ResourceInjector} instance. Will first call the {@link TransienceValidatingResourceInjector}, to
+     * ensure the fixture's approach of injecting the default classes (like the {@link EventBus} and {@link CommandBus}
+     * for example) is maintained. Afterward, the custom {@code ResourceInjector} provided through the {@link
+     * #registerResourceInjector(ResourceInjector)} is called. This will (depending on the implementation) inject more
+     * resources, as well as potentially override resources already injected by the {@code
+     * TransienceValidatingResourceInjector}.
+     */
+    private class WrappingResourceInjector implements ResourceInjector {
```

Comment:
Class looks like it could be static

Reply:
This requires making the `TransienceValidatingResourceInjector` static too, but fair argument nonetheless. Will do